proposals/extensions.md

```diff
@@ -1,30 +1,58 @@
 # Extension members
 
-## What's changed
-
-This proposal is an update on [Anonymous extension declarations](https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/extensions/anonymous-extension-declarations.md) in the following ways:
-
-- It leans into a parameter-style syntax for specifying the receiver.
-- It includes syntax for generating compatible extension methods.
-- It clarifies the goals of lowering, while leaving more details to implementation.
-
-### Receiver specification: `this` vs parameter
-
- The strongest sentiment I've heard is that whichever approach we take should be done *consistently*. Of the two there's a lean towards the parameter approach. This proposal leans into the parameter-based approach as strongly as it can.
-
-### Compatibility with classic extension methods
-
-There is a clear desire to be able to move classic extension methods forward into new syntax without breaking existing callers. This proposal lets extension methods be marked for compatibility, which will cause them to generate visible static methods in the pattern of classic extension methods.
-
-### Lowering
+## Declaration
 
-The proposal takes the stance that the declarations generated from lowering - static methods and/or types - should be hidden from the language level, making extension members a "real" abstraction. The implementation strategy needs to establish metadata and rules that can be followed and consumed by other compilers, and that ensure stability and compatibility for consuming code as APIs evolve.
+### Syntax
+
+```antlr
+class_body
+    : '{' class_member_declaration* '}' ';'?
+    | ';'
+    ;
+
+class_member_declaration
+    : constant_declaration
+    | field_declaration
+    | method_declaration
+    | property_declaration
+    | event_declaration
+    | indexer_declaration
+    | operator_declaration
+    | constructor_declaration
+    | finalizer_declaration
+    | static_constructor_declaration
+    | type_declaration
+    | extension_container // add
+    ;
+
+extension_container // add
+    : 'extension' type_parameter_list? '(' receiver_parameter ')' type_parameter_constraints_clause* extension_body
+    ;
+
+extension_body // add
+    : '{' extension_member_declaration* '}'
+    ;
+
+extension_member_declaration // add
+    : method_declaration
+    | property_declaration
+    | indexer_declaration
+    | operator_declaration
+    | type_declaration
+    ;
+
+receiver_parameter // add
+    : attributes? parameter_modifiers? type identifier?
+    ;
+```
 
-## Declaration
+Extension containers shall only be declared in non-generic, non-nested static classes.  
+It is an error for a type to be named `extension`.  
```

Comment:
> It is an error for a type to be named `extension`.

Is it because it would be ambiguous with an instance constructor declaration like this?

```cs
class extension
{
   extension() { }
}
```

But only static classes can contain `extension`s, and static classes cannot have instance constructors.

Reply:
This is not because of possibly ambiguity. It will simplify parsing and improve error recovery. We've been moving towards reserving names for new contextual keywords we introduce (`scoped`, `record`, `file`, `required`). 
We've made all-lowercase-alpha names a warning wave warning as a preparation step. Now (with `field`) we've introduced a process for introducing such breaks.

Reply:
Can we note that this error is tied to language version?